packages/common/src/ContractInteractor.ts

```diff
@@ -462,20 +445,20 @@ export class ContractInteractor {
     return this.relayCallMethod(_.maxAcceptanceBudget, _.relayRequest, _.signature, _.approvalData).encodeABI()
   }
 
-  async getPastEventsForHub (extraTopics: string[], options: PastEventOptions, names: EventName[] = ActiveManagerEvents): Promise<EventData[]> {
+  async getPastEventsForHub (extraTopics: Array<string[] | undefined>, options: PastEventOptions, names: EventName[] = ActiveManagerEvents): Promise<EventData[]> {
```

Comment:
This is not the real type
its actually `Array<string[] | string | undefined>`, to avoid shotgun - modification on every place that did `[topic]`
(e.g. here https://github.com/opengsn/gsn/pull/704/files#diff-99b458921a807bec203ad61c942975b0ddf9aef32a42c9ec267a85dec986bfdcR936)

(note that we never check the value of this type - it is passed as-is into getPastEvents, which uses web3 type

```
export interface LogsOptions {
    fromBlock?: BlockNumber;
    address?: string | string[];
    topics?: Array<string | string[] | null>;
}
```
(and BTW - its the first time I see that even `address` can have multiple values.... Might be useful sometimes in the future)

Reply:
It is not passed as-is - event name topic is added:

```
    const eventTopic = event2topic(contract, names)
    topics.push(eventTopic)
    if (extraTopics.length > 0) {
      topics.push(...extraTopics)
    }
```

I am ok with forcing all places that use this method to add `[ x ]` around the topic they want, if that is ok with you.

Reply:
What I mean is:
eventTopic is a single topic, representing the log itself.
extraTopics is array of topics.
each TOPIC (either the eventTopic or any item of extraTopics) is `string|null|string[]`
with your model, you force each of these topics into `string[]`
in most cases, topics are strings (or null, for wildcard) and its easier to understand them that way.
in rare cases we need multiple values, in which case we need an array of string.
That's the API defined by getPastEvents. You redefine it as "always arrays", even for the null values.

I think its a bit weird.

Reply:
Actually, I'm debugging in issue on kovan, which a server returned me an error:
`{code: -32600, message: 'data type size mismatch, expected 32 got 0'}`
I just confirmed that the root cause is passing into the topics array an element `"[]"`
(hardhat/ganache might work OK with it, and maybe even geth (didn't check) but kovan's OpenEthereum rejects it..)